CHANGELOG-3.6.md

```diff
@@ -8,6 +8,10 @@ Previous change logs can be found at [CHANGELOG-3.5](https://github.com/etcd-io/
 
 See [code changes](https://github.com/etcd-io/etcd/compare/v3.5.0...v3.6.0).
 
+### Breaking Changes
+
+- `etcd` will no longer start on data dir created by newer versions (for example etcd v3.6 will not run on v3.7+ data dir). To downgrade data dir please check out `etcdutl migrate` command.
```

Comment:
> will no longer start on data dir created by newer versions (for example etcd v3.6 will not run on v3.7+ data dir).

Would we need etcdutl 3.7 to perform the downgrade or etcdutl-3.6 would handle this ? 

Thinking about scenario: 
  - we attempted upgrade of clusters.  It failed. Some logic decided to restart from previous version of 'image'. 
  - the new image contains etcdutl-3.6 and etcd 3.6
  - the image cannot contain self-healing logic. 
 
TBH: A single minor version downgrades are pretty important from reliability perspective.

Reply:
This is a good point, the current storage versioning proposes a solution, but it might be worth discussing more details here.

When talking about storage downgrades it's important to look in context of whole downgrade process. As etcd WAL can contain new etcd entries, etcd downgrade cannot be done ad hock. It requires a whole procedure to ensure that all members are aware of downgrade and prepare for it. 

Skipping this procedure and just starting older etcd on the same data will always come with a risk encountering unknown WAL entry type. The restriction for downgrading is introduced to guide users towards the safer process as default. Still setup you described is pretty common in kubernetes and we should support it.

By default downgrading v3.7 will require using etcdutl v3.7, this way the tool with full information can make sure that the downgrade is correct and safe, still downgrading using older version is possible, but not as safe. This is why `etcdutl migrate --force` was introduced. In this mode older etcdutl will be can downgrade newer version, based on assumption that etcd bbolt schema is backward compatible. Additionally it should purge all the unknown fields and WAL entries to make. As this will strongly depend on backward compatibility between one minor version, we should add dedicated testing that we will not break it by mistake.

Does using `etcdutl migrate --force` in this scenario address your concern?

Reply:
The previous experience was ~safe  because of 2-release. In release N+1 we added support for feature and 'experimental flags' activating it, but by default we were not writing WAL logs that couldn't be interpreted by N release. 

I agree that N+2 releases is killing velocity. I'm thinking whether we can divide lifecycle of the N+1 server into 2 phases: 
   - pre cluster upgrade (no new features are being used)
   - post cluster upgrade (quorum agreed to take new version) when new WAL logs are accepted.
 I think it's how it was described in the design. 

So maybe the Changelog phrasing that when quorum commits cluster upgrade than explicit downgrade is being needed. 

This is big change and we need @gyuho approval.